app/controllers/categories_controller.rb

```diff
@@ -13,11 +13,36 @@ def create
     end
   end
 
+  def index
+    @categories = Category.all
+  end
+
   def show
     @category = Category.find(params[:id])
     @jobs = (@category.jobs).sort_by { |job| job.title}
   end
 
+  def edit
+    @category = Category.find(params[:id])
+  end
+
+  def update
+    @category = Category.find(params[:id])
+    if @category.update_attributes(category_params)
+      redirect_to categories_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    category = Category.find(params[:id])
+    jobs = category.jobs
+    category.terminate_jobs(jobs)
```

Comment:
you don't need your own method here. You can user AR. `category.jobs.destroy_all`